bats/send_metrics.bats

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bats
+
+setup() {
+    load $BATS_TEST_DIRNAME/helper/common.bash
+    export PATH=$PATH:~/go/bin
+    export NOMS_VERSION_NEXT=1
+    cd $BATS_TMPDIR
+    mkdir "dolt-repo-$$"
+    cd "dolt-repo-$$"
+    dolt init
+    dolt config --global --unset metrics.disabled
+}
+
+teardown() {
+    rm -rf "$BATS_TMPDIR/dolt-repo-$$"
+    rm -rf "$BATS_TMPDIR/config-$$/.dolt/eventsData/"
+}
+
+@test "test event flush locking" {
+    # copy test event files to appropriate dir
+    cp -a $BATS_TEST_DIRNAME/helper/testEvents/ $BATS_TMPDIR/config-$$/.dolt/eventsData/
+
+    # kick off two child processes, one should lock the other out of the events dir
+    dolt send-metrics -output >file1.txt &
+    dolt send-metrics -output >file2.txt &
+
+    # wait for processes to finish
+    wait
+
+    # get the line count of each output file
+    event_count1=`wc -l file1.txt`
+    event_count2=`wc -l file2.txt`
+  
+    if [[ "$event_count1" = *"4 file1.txt" ]] && [[ "$event_count2" = *"0 file2.txt" ]]; then
+      return 0
+    fi
+    if [[ "$event_count1" = *"0 file1.txt" ]] && [[ "$event_count2" = *"4 file2.txt" ]]; then
+      return 0
+    fi
+
+    return 1
+}
+
```

Comment:
New and improved test.. It's the cleaned up version of the commented out test below. Basically I removed the additional if else logic since we are suppressing the status codes from the wait command and just checking the file line length yet.